internal/namespace/configreload_test.go

```diff
@@ -490,13 +489,13 @@ func TestConfigListenerE2E(t *testing.T) {
 	err = nm.Start()
 	assert.NoError(t, err)
 
+	waitInit.Wait()
+
+	waitInit = namespaceInitWaiter(t, nmm, []string{"ns3"})
+
 	err = ioutil.WriteFile(configFilename, []byte(exampleConfig2extraNS), 0664)
 	assert.NoError(t, err)
 
-	for nm.namespaces["ns3"] == nil {
```

Comment:
This wait was unnecessary after `namespaceInitWaiter` was introduced, and I took the opportunity to eradicate a `time.Sleep` loop